### DISCUSSION TITLE
How to persist multiple localizations at once?

### DISCUSSION THREAD

**Discussion by luca-nardelli** [payloadcms/payload#2421] [Q&A]
Hi all! 

I was looking into the localization solution that Payload offers out of the box, however I cannot seem to find a way to persist multiple languages at once within the same API call.

If I try to push the object representation of the field (e.g. `{it: '...', en: '...'}`) instead of the field value, I either get a mongoose validation error if I don't specify the locale or I don't get anything persisted if I set `locale: 'all'`. 

Does anyone know how to achieve this? I wanted to run some bulk imports of data and I would like to avoid sending 1 update request for every locale

**Comment by DanRibbens** [comment]
Hey @luca-nardelli, saving multiple locales is on our radar, but I can't say when it will be available.

> I wanted to run some bulk imports of data and I would like to avoid sending 1 update request for every locale

To avoid this, you'll need to use the mongoose models directly instead of Payload's APIs until we can build it in.

Does that help?

**Comment by luca-nardelli** [comment]
It does! However, how do I access the models directly? I tried looking for this in the docs but did not find pointers. Thanks!

**Reply by DanRibbens** [reply]
Awesome! 

From your payload instance you have access to collections which is a keyed object using the collecction config slug and has those include the mongoose Models.

Instead of calling `payload.create()` directly, do something like this: `payload.collections[collectionSlug].Model.create(docWithLocales)`.

**Comment by luca-nardelli** [comment]
Didn't know that, cool! Thanks for the help, will try later. Is this something that could be added to the docs? If so, I can see if I have some time to write something down.

**Reply by DanRibbens** [reply]
Yeah, it could go in the docs!

If you want to make a PR, I'd suggest adding a new section to [collections](https://payloadcms.com/docs/configuration/collections) and a similar one for [globals](https://payloadcms.com/docs/configuration/globals). I would make a new section `###Database model` (or similar) and talk about when you might need to use the database model directly, how to access the `Model` from `payload` and then link to Mongoose docs for what is available.

Also please add a cautiontionary `<banner type="warning" />` that warns developers that raw database access will bypass all hooks, access control and validation that payload normally handles.

**Comment by 60pfennig** [comment]
I'm dealing with multiple languages right now and I have to say that a bulk update of all locales is a very important feature. Without a bulk edit you will run into inconsistencies. If fields are required AND localized the current system won't enforce the right localization. Instead it relies on the fallback language mechanism which just provides the wrong translation. There should be at least a flag in the config which enforces a strict localization. A document should only save if every localized field is translated. The same goes for the API. It should be possible or needed (if strict translation turned on) that if a document is updated/created every translation is mandatory and there needs to be a way to pass these in one API call. If a translation is missing Payload should reject the operation completly. Without these behaviour there will be data which don't fullfill the contract of the type. If you fetch a language which wasn't provided but the field is required and therefor it is not optional and has to be set in the right language.

**Comment by chrisvanmook** [comment]
Do we have any updates on this for v3? I have a similar problem:

I'm writing an export / import feature, where I want to export a collection, to eventually import this on another environment. This collection has localized fields, so I initially wanted to do something like this:

Having a collection with a localized field:
```ts
...
fields: [
  {
    name: 'question',
    label: 'Question',
    type: 'text',
    localized: true,
    required: true,
  },
...
```

```ts
const posts = await payload.find({
  collection: 'posts',
  pagination: false,
  locale: 'all',
})
```

Note that now question isn't of type string, but an object like:
```ts
{
question: {
    en: 'Hello',
    nl: 'Hallo',
  }
}
```

To eventually import it:
```ts
await payload.create({
  collection: 'posts',
  data: posts, // containing the localized question field
});
```

This isn't possible now. I could of course iterate through all the available locales, and create one and then update the entry for the remaining languages, but this feels very cumbersome. `payload.collections[collectionSlug].Model.create(docWithLocales)` does not exist for v3 it seems... I'm using postgres by the way.

**Reply by DanRibbens** [reply]
Correct `payload.collections[slug].Model` was only for mongodb. You can use the DB adapter methods directly. The adapter methods retrieve and accept data objects that are keyed by locale.
This should give you what you need to have a way to read/write all locales at the same time:
```ts
await payload.db.create({
  collection,
  req,
  data: {
   // localized example: 
    title: {
      en: 'English Title,
      fr: 'French Title', // etc.
    },
   // not localized property example:
   slug: 'abc',
})
```

Note, if you have versions enabled on the collection, you will want to call payload.update on the same document after it comes back from create to make it have a versions record also or call payload.db.createVersion directly.

This should help in the interim until we have an established local API way to make updates/creates on multiple locales at once.

Cheers!

**Reply by chrisvanmook** [reply]
Thanks for the quick reply! Awesome, this will do for now!

**Reply by Oudwins** [reply]
@DanRibbens The rest api doesn't support this yet right? At least not for v2

**Reply by hbkrunal** [reply]
@DanRibbens   do you know if there is method to update multiple localizations at once similar to  `payload.db.create` ?

**Reply by Oudwins** [reply]
I also tried payload.db.create and could not get it to work for some reason. Need to try it further
